napari/layers/_shapes_layer/model.py

```diff
@@ -1571,86 +1514,153 @@ def on_mouse_release(self, event):
+
+    def _select_all(self):
+        if self.mode in (Mode.DIRECT, Mode.SELECT):
+            self.selected_shapes = list(range(len(self.data.shapes)))
+            self._set_highlight()
+
+    def _finish_drawing(self):
+        """Reset properties used in shape drawing so new shapes can be drawn.
+        """
+        index = copy(self._moving_shape)
+        self._is_moving = False
+        self.selected_shapes = []
+        self._drag_start = None
+        self._drag_box = None
+        self._is_selecting = False
+        self._fixed_vertex = None
+        self._moving_shape = None
+        self._moving_vertex = None
+        self._hover_shape = None
+        self._hover_vertex = None
+        if self._is_creating is True and self.mode == Mode.ADD_PATH:
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+            else:
+                self.data.edit(index, vertices[:-1])
+        if self._is_creating is True and self.mode == Mode.ADD_POLYGON:
+            vertices = self.data._vertices[self.data._index == index]
+            if len(vertices) <= 2:
+                self.data.remove(index)
+        self._is_creating = False
+        self.refresh()
+
+    def remove_selected(self):
+        """Remove any selected shapes.
+        """
+        # TODO: can this be made private?
+        to_remove = sorted(self.selected_shapes, reverse=True)
+        for index in to_remove:
+            self.data.remove(index)
+        self.selected_shapes = []
+        shape, vertex = self._shape_at(self._cursor_coord)
+        self._hover_shape = shape
+        self._hover_vertex = vertex
+        self.status = self.get_message(self._cursor_coord, shape, vertex)
+        self.refresh()
+
+    def _hold_to_pan_zoom(self):
+        if self.mode != Mode.PAN_ZOOM:
+            # on key press
+            prev_mode = self.mode
+            prev_selected_shapes = copy(self.selected_shapes)
+            self.mode = Mode.PAN_ZOOM
+
+            yield
+
+            # on key release
+            self.mode = prev_mode
+            self.selected_shapes = prev_selected_shapes
+            self._set_highlight()
+
+    def _hold_to_move_shape(self):
+        # on key press
+        self._fixed_aspect = True
+        box = self._selected_box
+        if box is not None:
+            size = box[Box.BOTTOM_RIGHT]-box[Box.TOP_LEFT]
+            if not np.any(size == np.zeros(2)):
+                self._aspect_ratio = abs(size[1] / size[0])
+            else:
+                self._aspect_ratio = 1
+        else:
+            self._aspect_ratio = 1
+        if self._is_moving:
+            self._move(self._cursor_coord)
+
+        yield
+
+        # on key release
+        self._fixed_aspect = False
+        if self._is_moving:
+            self._move(self._cursor_coord)
+
+    default_keybindings = {
+        'Space': _hold_to_pan_zoom,
+        'Shift': _hold_to_move_shape,
```

Comment:
`Shift` doesn't seem to be working, i.e. select a shape and start resizing - the aspect ratio is no longer fixed